db/migrate/20170410185513_create_checkouts.rb

```diff
@@ -0,0 +1,9 @@
+class CreateCheckouts < ActiveRecord::Migration[5.0]
+  def change
+    create_table :checkouts do |t|
+      t.timestamps null: false
+      t.references :book, index: true, foreign_key: true, null: false
+      t.references :user, index: true, foreign_key: true, null: false
```

Comment:
@derekprior, This is a very good point. I need to think about it, but I should have a solution shortly.